src/librustc_middle/mir/interpret/mod.rs

```diff
@@ -549,18 +549,20 @@ pub fn write_target_uint(
     mut target: &mut [u8],
     data: u128,
 ) -> Result<(), io::Error> {
-    let len = target.len();
     match endianness {
-        Endian::Little => target.write_uint128::<LittleEndian>(data, len),
-        Endian::Big => target.write_uint128::<BigEndian>(data, len),
-    }
+        Endian::Little => target.write(&data.to_le_bytes())?,
+        Endian::Big => target.write(&data.to_be_bytes())?,
+    };
+    Ok(())
 }
 
 #[inline]
 pub fn read_target_uint(endianness: Endian, mut source: &[u8]) -> Result<u128, io::Error> {
+    let mut buf = [0; 16];
```

Comment:
```suggestion
    let mut buf = [0u; mem::size_of::<u128>()];
```

Reply:
Whoops, needs `std::mem::size_of::<u128>()`.
Good idea though, even if... well, I don't _expect_ the essential nature of numbers to change, but it does make things explicit.